vite.config.js

```diff
@@ -1,28 +1,258 @@
+              console.log(name, iterator.exclude);
+            }
+            if(!iterator.exclude.includes(name)){
+              const entry = this.emitFile({
+                type: 'chunk',
+                id: file,
+                fileName: file.replace(iterator.replaceExpression, baseDirectory),
+              });
+              _entryPoints.push(entry);  
+            }else{
+              console.log('******* EXCLUDING ***********', name);
+            }
+          }
+        } else {
+          const entry = this.emitFile({
+            type: 'chunk',
+            id: iterator.path,
+            fileName: iterator.name,
+          });
+          _entryPoints.push(entry);
+        }
+      }
+    },
+    generateBundle(options, bundle) {
+      /** testing exports */
+
+      if (process.env.OUTPUT_EXPORTS) {
+        const exports = {
+        };
+
+        for (const chunk of Object.keys(bundle)) {
+          if (bundle[chunk].exports) {
+            for (const _export of bundle[chunk].exports) {
+              exports[_export] = chunk;
+            }
+          }
+        }
+
+        fs.writeFileSync('dist/dependencies.json', JSON.stringify(exports, null, 4));
+        fs.writeFileSync('dist/actualBundle.json', JSON.stringify(bundle, null, 4));
+      }
+      return null;
+    },
+    // async transform(code, id) {
+    //   const avoidTransform = false;
+
+    //   if (code === null || id === null) {
+    //     return null;
+    //   }
+
+    //   const loader = path.parse(id).ext.replace('.', '');
+
+    //   const isNodeModule = id.includes('node_modules');
+
+    //   if (!esbuildLoaders.includes(loader)) {
+    //     return {
+    //       code: code,
+    //       map: null,
+    //     };
+    //   }
+
+    //   const transformed = await transform(code, {
+    //     loader: loader,
+    //     format: loader === 'js' && !isNodeModule ? 'esm' : undefined,
+    //     minifySyntax: false,
+    //     minifyWhitespace: true,
+    //     keepNames: true,
+    //     sourcemap: true,
+    //     target: ['es6'],
+    //   });
+
+    //   return {
+    //     code: transformed.code,
+    //     map: transformed.map,
+    //   };
+    // },
+    resolveImportMeta(property, {moduleId}) {
+      /** Send force null to avoid import.meta transformation */
+      return null;
+    },
+  };
+};
+
+/** Use totum if not production */
+plugins = process.env.NODE_ENV !== 'production' ? plugins.concat([metaversefilePlugin()]) : plugins.concat([
+  build(),
+]);
+
+/** Vite config for production */
+const viteConfiProduction = {
+  build: {
+    polyfillModulePreload: false,
+    format: 'es',
+    target: 'esnext',
+    manifest: true,
+    rollupOptions: {
+      preserveEntrySignatures: 'strict',
+      output: {
+        // __vite_skip_esbuild__: true,
+        exports: 'named',
+        minifyInternalExports: false,
+        format: 'es',
+        manualChunks: id => {
+          if (id.includes('three/build')) {
```

Comment:
These should not be config exceptions.

Reply:
manual chunking is removed. Only kept for metaverse-modules.